log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java

```diff
@@ -323,10 +325,19 @@ public void start() {
         LOGGER.info("Starting configuration {}...", this);
         this.setStarting();
         if (watchManager.getIntervalSeconds() >= 0) {
-            LOGGER.info(
-                    "Start watching for changes to {} every {} seconds",
-                    getConfigurationSource(),
-                    watchManager.getIntervalSeconds());
+            if (uris != null && uris.size() > 0) {
+                LOGGER.info(
+                        "Start watching for changes to {} and {} every {} seconds",
+                        getConfigurationSource(),
+                        uris,
+                        watchManager.getIntervalSeconds());
+                watchManager.addMonitorUris(configurationSource.getSource(), uris);
```

Comment:
Can't we instead for each URI create a new `ConfigurationFileWatcher` and register it using `watchManager.watch(cfgSource, watcher)`?